### ISSUE TITLE
gtklock package does not set up PAM config, making it impossible to unlock the screen

### ISSUE THREAD

**Issue by rollcat** [NixOS/nixpkgs#240886]
### Describe the bug

The [gtklock package](https://github.com/NixOS/nixpkgs/tree/nixos-23.05/pkgs/tools/wayland/gtklock), when installed system-wide, does not drop any configuration into `/etc/pam.d/gtklock`. Upon first installation, user trying out the program will be unable to unlock the screen, without taking additional configuration steps (which might be non-obvious).

### Steps To Reproduce

1. Install the package, e.g. add it to `environment.systemPackages`.
2. Run `gtklock` while in a Wayland session (e.g. Sway).
3. Observe that typing your login password does not unlock the screen.

The screen can still be unlocked by switching to another VT (unless switching is disabled), or by logging in via SSH, and executing `pkill gtklock`.

### Expected behavior

In a default installation (e.g. when defining `programs.gtklock.enable = true`, which currently does not exist), it should be possible to use whichever standard mechanisms are present on the system for unlocking the screen (typing a password, using the fingerprint reader, etc).

### Additional context

Workaround:

```nix
security.pam.services.gtklock.text = lib.readFile "${pkgs.gtklock}/etc/pam.d/gtklock";
```

### Notify maintainers

@dit7ya @r-ryantm

### Metadata

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.35, NixOS, 23.05 (Stoat), 23.05.1353.35130d4b4f0`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(root): `"nixos-23.05, nixos-hardware, nixos-unstable"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by dit7ya** [comment]
The workaround is mentioned in the package's longDescription field - https://github.com/NixOS/nixpkgs/blob/b72aa95f7f096382bff3aea5f8fde645bca07422/pkgs/tools/wayland/gtklock/default.nix#L50

I am not sure what would be a better solution?

**Comment by Shawn8901** [comment]
I guess the best solution would be to create a module (like suggested in the expected behavior).
The module would also add the required PAM settings if enabled. That's also how other programs do that, if additional setup is neeeded

**Comment by rollcat** [comment]
> The workaround is mentioned in the package's longDescription field -
> [...]
> I am not sure what would be a better solution?

You're correct! I totally overlooked that. I usually go looking for `programs.*.enable`, which I think would be the most user-friendly way to install the package and have a minimal working setup.

Having `programs.gtklock.plugins = []` would also be a nice way to enable extra plugins, or specify other configuration (the program is already searching `$XDG_CONFIG_DIRS`).
